com.ibm.ws.logging.hpel: java.lang.IllegalArgumentException: Unable to parse --minDate value on IBM Public Cloud

RTC 279159

Our new Public Cloud prototype build seems to consistently fail on the same bucket: unit_com.ibm.ws.logging.hpel

Build: OpenLiberty external contributor build on IBM public cloud 20201026-0516
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_x1w3IRc_EeuFUN9G2T7ZeA

Junit: https://libfsfe06.hursley.ibm.com/liberty/dev/Xo/cont/20201026-0516-_x1w3IRc_EeuFUN9G2T7ZeA/junits/unit_com.ibm.ws.logging.hpel/42093383999e5fe07119ffc5ac376701eda6597422b6be796797c73c87ca442d/TESTS-TestSuites.xml

```
java.lang.IllegalArgumentException: Unable to parse --minDate value. 
    at com.ibm.ws.logging.hpel.viewer.BinaryLog.setStartDate(BinaryLog.java:511) 
    at com.ibm.ws.logging.hpel.viewer.BinaryLog$1.setValue(BinaryLog.java:265) 
    at com.ibm.ws.logging.hpel.viewer.BinaryLog$OneArgOption.accept(BinaryLog.java:227) 
    at com.ibm.ws.logging.hpel.viewer.BinaryLog.parseOptions(BinaryLog.java:418) 
    at com.ibm.ws.logging.hpel.viewer.BinaryLog.parseCmdLineArgs(BinaryLog.java:374) 
    at com.ibm.ws.logging.hpel.viewer.BinaryLogTest.helpTestParseCmdLineArgs(BinaryLogTest.java:450) 
    at com.ibm.ws.logging.hpel.viewer.BinaryLogTest.testParseCmdLine(BinaryLogTest.java:388) 
    at com.ibm.ws.logging.hpel.viewer.BinaryLogTest.testParseCmdLine(BinaryLogTest.java:400) 
    at com.ibm.ws.logging.hpel.viewer.BinaryLogTest.testTimePattern(BinaryLogTest.java:279) 
    at com.ibm.ws.logging.hpel.viewer.BinaryLogTest.testParseCmdLineArgsViewWithDates(BinaryLogTest.java:203) 
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) 
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) 
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) 
    at java.base/java.lang.reflect.Method.invoke(Method.java:566) 
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:45) 
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15) 
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:42) 
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20) 
    at test.common.SharedOutputManager$1.evaluate(SharedOutputManager.java:620) 
    at org.junit.rules.RunRules.evaluate(RunRules.java:18) 
    at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263) 
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68) 
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47) 
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231) 
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60) 
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229) 
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50) 
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222) 
    at org.junit.runners.ParentRunner.run(ParentRunner.java:300) 
    at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110) 
    at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58) 
    at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38) 
    at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62) 
    at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51) 
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) 
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) 
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) 
    at java.base/java.lang.reflect.Method.invoke(Method.java:566) 
    at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36) 
    at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24) 
    at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33) 
    at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94) 
    at com.sun.proxy.$Proxy2.processTestClass(Unknown Source) 
    at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:118) 
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) 
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) 
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) 
    at java.base/java.lang.reflect.Method.invoke(Method.java:566) 
    at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36) 
    at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24) 
    at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182) 
    at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164) 
    at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:413) 
    at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64) 
    at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48) 
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) 
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) 
    at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56) 
    at java.base/java.lang.Thread.run(Thread.java:836) 
Caused by: java.text.ParseException: Unparseable date: "2013-02-24, 11:12:13:999 UTC" 
    at java.base/java.text.DateFormat.parse(DateFormat.java:395) 
    at com.ibm.ws.logging.hpel.viewer.BinaryLog.setStartDate(BinaryLog.java:498) ... 58 more
```

--- a/dev/com.ibm.ws.logging.hpel/test/com/ibm/ws/logging/internal/hpel/LocaleUtils.java
+++ b/dev/com.ibm.ws.logging.hpel/test/com/ibm/ws/logging/internal/hpel/LocaleUtils.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2013 IBM Corporation and others.
+ * Copyright (c) 2013,2020 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at

--- a/dev/com.ibm.ws.logging.hpel/test/com/ibm/ws/logging/internal/hpel/LocaleUtils.java
+++ b/dev/com.ibm.ws.logging.hpel/test/com/ibm/ws/logging/internal/hpel/LocaleUtils.java
@@ -324,7 +324,8 @@ public class LocaleUtils {
             || currentLocale.equalsIgnoreCase("hi_IN") || currentLocale.equalsIgnoreCase("pa_IN")
             || currentLocale.equalsIgnoreCase("mr_IN") || currentLocale.equalsIgnoreCase("kn_IN")
             || currentLocale.equalsIgnoreCase("pa") || currentLocale.equalsIgnoreCase("mr")
-            || currentLocale.equalsIgnoreCase("kn") || currentLocale.equalsIgnoreCase("zh_SG"))
+            || currentLocale.equalsIgnoreCase("kn") || currentLocale.equalsIgnoreCase("zh_SG")
+            || currentLocale.equalsIgnoreCase("c.u_US"))
             return false;
         else
 

